Add metrics subsystem

waitinfuture opened the pull request.

TanYuxin-tyx commented on dev/checkstyle.xml:
```diff
@@ -0,0 +1,140 @@
+<!DOCTYPE module PUBLIC
```
Missing license

TanYuxin-tyx commented on pom.xml:
```diff
@@ -35,6 +35,17 @@
     <maven.compiler.target>${java.version}</maven.compiler.target>
     <maven.version>3.5.4</maven.version>
+    <build.testJarPhase>prepare-package</build.testJarPhase>
+    <CodeCacheSize>512m</CodeCacheSize>
```
Where is CodeCacheSize used? I don't get it.

TanYuxin-tyx commented on pom.xml:
```diff
@@ -35,6 +35,17 @@
+    <scala.binary.version>2.12</scala.binary.version>
+    <slf4j.version>1.7.16</slf4j.version>
+    <guava.version>14.0.1</guava.version>
+    <codahale.metrics.version>3.2.6</codahale.metrics.version>
```
3.2.6 is a very old version and why do we use this version? The new version is 4.2.10
Ref: https://github.com/dropwizard/metrics/releases?page=1
https://metrics.dropwizard.io/4.2.0/getting-started.html

TanYuxin-tyx commented:
Thanks @waitinfuture for the PR.👍🏻  I have some minor comments and please have a look when you are free.

waitinfuture commented:
> Thanks @waitinfuture for the PR.👍🏻 I have some minor comments and please have a look when you are free.

Hi @TanYuxin-tyx  I have addressed the comments, please take a look. Thanks!
